### PR TITLE
Perform retries in APIClient when encountering EventualConsistency errors

### LINKED ISSUE #585: API client retry when hit specific dependency errors
* create a new error code in L0
* when client encounters it, retry X(3) times, with no additional wait time.

### PR DESCRIPTION
**Pull request by zpatrick**
** What does this pull request do?**
This adds retry logic to the `client` package whenever the `EventualConsistencyError` code is encountered. In that scenario, the request is retried up to 3 times. 

**How should this be tested?**
Unit Tests

**Checklist**
- [x] Unit tests
- ~[ ] Smoke tests (if applicable)~
- ~[ ] System tests (if applicable)~
- ~[ ] Documentation (if applicable)~

closes #585

### CODE REVIEW

File: client/api_client.go [hunk 1]
```diff
@@ -11,9 +11,12 @@ import (
 
 	"github.com/quintilesims/layer0/common/errors"
 	"github.com/quintilesims/layer0/common/models"
+	"github.com/quintilesims/layer0/common/retry"
 	"github.com/zpatrick/rclient"
 )
 
+const MaxEventualConsistencyRetries = 3
```
Comment (nehayward) [inline client/api_client.go]: [nit] `MaxRetries` Should suffice here.

File: client/api_client.go [hunk 2]
```diff
@@ -34,7 +37,8 @@ func NewAPIClient(c Config) *APIClient {
 		httpClient = &http.Client{Transport: tr}
 	}
 
-	doer := newDebugRequestDoer(httpClient)
+	doer := wrapDebugRequestDoer(httpClient)
+	doer = wrapRetryRequestDoer(doer)
```
Comment (nehayward) [inline client/api_client.go]: Hmm this looks like it's overwrite the debug doer at first glance. Maybe simplify this. Also it's a bit confusing that the function names are the same but one takes a client and the other takes a doer.

## Option 1
```
debugDoer := wrapDebugRequestDoer(httpClient)
doer := wrapRetryRequestDoer(debugDoer)
```

## Option 2
```
doer := wrapRetryRequestDoer(wrapDebugRequestDoer(httpClient))
```

## Option 3
```
debugDoer := wrapDebugRequestDoer(httpClient)
rertyDoer := wrapRetryRequestDoer(httpClient)
doer := newDoer(doer ...rclient.RequestDoer)
```

File: client/api_client.go [hunk 3]
```diff
@@ -100,3 +129,15 @@ func newResponseReader() rclient.ResponseReader {
 		return nil
 	}
 }
+
+func readServerError(resp *http.Response) *errors.ServerError {
+	defer resp.Body.Close()
+
+	var se models.ServerError
```
Comment (nehayward) [inline client/api_client.go]: [nit] maybe `serverErr` or `errS`

File: common/errors/code.go [hunk 4]
```diff
@@ -16,5 +16,5 @@ const (
 	ServiceDoesNotExist      ErrorCode = "ServiceDoesNotExist"
 	TaskDoesNotExist         ErrorCode = "TaskDoesNotExist"
 	UnexpectedError          ErrorCode = "UnexpectedError"
-	FailedRequestTimeout     ErrorCode = "FailedRequestTimeout"
+	EventualConsistencyError ErrorCode = "EventualConsistencyError"
```
Comment (nehayward) [inline common/errors/code.go]: Good name 👍

File: client/api_client.go (hunk 2 above)
Comment (zpatrick) [reply client/api_client.go]: Ok, I'll go w/ option 1

**Pull request merged**